src/leo_gateway_http_commons.erl

```diff
@@ -860,22 +894,22 @@ get_range_object_large( Req, Bucket, Key, Start, End, Total, Index, CurPos) ->
 
 %% @doc
 %% @private
-send_chunk(_Req,_,_Key, Start,_End, CurPos, ChunkSize) when (CurPos + ChunkSize - 1) < Start ->
+send_chunk(_Req,_,_Key, Start,_End, CurPos, ChunkSize, _Socket, _Transport) when (CurPos + ChunkSize - 1) < Start ->
     %% skip proc
     CurPos + ChunkSize;
-send_chunk(Req,_Bucket, Key, Start, End, CurPos, ChunkSize) when CurPos >= Start andalso
+send_chunk(_Req,_Bucket, Key, Start, End, CurPos, ChunkSize, Socket, Transport) when CurPos >= Start andalso
                                                                  (CurPos + ChunkSize - 1) =< End ->
     %% whole get
     case leo_gateway_rpc_handler:get(Key) of
         {ok, _Meta, Bin} ->
             %% @FIXME current impl can't handle a file which consist of grand children
             %% ?access_log_get(Bucket, Key, ChunkSize, ?HTTP_ST_OK),
-            cowboy_req:chunk(Bin, Req),
+            Transport:send(Socket, Bin),
```

Comment:
This is an existing bug not handling errors returned by Transport:send|cowboy_req:chunk.
In this case, We need to handle this request as a error but we already sent the http header and a part of body,
What should we do? 
The answer is do `Transport:close`.

Since there are lots of errors to be handled as the same above, 
Please fix those bugs as well.